spec.emu

```diff
@@ -11,6 +11,45 @@ shortname: &lt;a href="https://github.com/tc39/proposal-immutable-arraybuffer"&g
 contributors: Mark S. Miller, Richard Gibson
 </pre>
 
+<emu-clause id="sec-ecmascript-data-types-and-values" number="6">
+  <h1>ECMAScript Data Types and Values</h1>
+
+  <emu-clause id="sec-data-blocks" number="2.9">
+    <h1>Data Blocks</h1>
+    <p>A data block that resides in memory that can be referenced from multiple agents concurrently is designated a <dfn variants="Shared Data Blocks">Shared Data Block</dfn>. A Shared Data Block has an identity (for the purposes of equality testing Shared Data Block values) that is <em>address-free</em>: it is tied not to the virtual addresses the block is mapped to in any process, but to the set of locations in memory that the block represents. Two data blocks are equal only if the sets of the locations they contain are equal; otherwise, they are not equal<ins>.</ins> <del>and the intersection of the sets of locations they contain is empty</del> <ins>The intersection of the sets of locations contained by two non-equal data blocks may be non-empty only when both data blocks are immutable and one is a strict subset of the other</ins>. Finally, Shared Data Blocks can be distinguished from Data Blocks.</p>
```

Comment:
> The "and one is a strict subset of the other" is clearly wrong because you could have two overlapping slices of a common parent.

Hmm, good point. But I still want text to point out that overlapping memory locations are not acceptable without an immutable common parent (e.g., a 16-byte data block at location X vs. a 16-byte data block at location X + 4). Please look over the latest text.

> > for the purposes of equality testing Shared Data Block values
> 
> Where?

I don't know; that's existing text that I don't want to touch here.

Reply:
Given @waldemarhorwat's point (which we agreed with at plenary) that the spec not state more than equivalence to `buf.slice(s,e).toImmutable()`, I don't think we need to touch the notion of shared locations at all. Let the spec imply that `buf.sliceToImmutable(s,e)` makes copies, just as `buf.slice(s,e).toImmutable()` would. When `buf` is also immutable, an implementation can share (i.e., "zero-copy") as an implementation optimization. But there's no reason for the spec to bless this beyond a non-normative note.

Then the "Shared Data Blocks" in the spec can remain as they are, and would continue to apply ***only*** to SABs.

Waddaya think?